Add overlay

tfc opened the pull request:
This PR makes the library useable as an overlay.

~~It also enables aarch64 runs with flake-utils.
I did however not test aarch64 as i don't have a strong aarch64-linux system handy.~~

My intention to have an overlay is more of 

- being able to use this without flakes
- inside a flake that already uses overlays, just inject this one too instead of having it import nixpkgs another time

Please let me know what you think.

tfc commented:
It seems like the aarch64 versions don't work because some packages are even unsupported. Please let me know if the overlay thingy is acceptable at all. If yes, then this is worth fixing by at least disabling what can't work in the CI and i'll do it.

zimbatm commented on flake.nix:
```diff
@@ -2,14 +2,23 @@
+        };
+      in
+      {
+        lib = pkgs.testers.legacyDistros;
```
let's also export the overlay then

zimbatm reviewed (commented):
sounds good! As long as the checks are passing it should be ok.

Can you also update the README?

picnoir commented:
I'm also fine with this, but we should drop the second commit. Quoting "archives.fedoraproject.org":

> Note: The Fedora releases here are no longer supported or maintained, so they do not receive bug fixes or security updates. We do not recommend using these releases any more. To obtain the latest, supported version of Fedora, please refer to the [main download page](http://get.fedoraproject.org/).

tfc commented:
> I'm also fine with this, but we should drop the second commit. Quoting "archives.fedoraproject.org":
> 
> > Note: The Fedora releases here are no longer supported or maintained, so they do not receive bug fixes or security updates. We do not recommend using these releases any more. To obtain the latest, supported version of Fedora, please refer to the [main download page](http://get.fedoraproject.org/).

I added this PR because the old URL doesnt work any longer. I am however happy to drop this commit and add either another one or rebase onto a fixed main branch.

tfc commented:
@zimbatm I added the overlay in the `overlays` attribute and dropped flake-utils all along as it currently only correctly evaluates on intel machines anyway.

@picnoir plz forget what i said about the download mirror. I swear, yesterday the old url gave me 404 errors and when i browsed to it, the readme in the server's folder refered me to the other URL. And today it all just works again. However, i dropped the commit and everything works fine.

tfc commented:
@zimbatm I am happy to update the documentation, although i am not sure how to do that best.
The README you mentioned does not really show how to use it all, so i would add an example under `docs/`? If yes, would you like an extra paragraph in the existing docs or its own document that contains an example with/without flakes that uses the overlay?

zimbatm commented:
yeah, I think an extra paragraph in the getting started section would be good enough. Flake is the main use-case, but by the way, if you're not, here is a side-note on how to use it.

tfc commented:
@zimbatm done

picnoir commented:
This should probably be in the reference, the getting started section is meant to be read from top to bottom in a interactive way.

But let's address that in a subsequent commit.

picnoir reviewed (approved):
LGTM, thanks!

picnoir commented:
> But let's address that in a subsequent commit.
